src/credentialstore/credentialstore.ts

```diff
@@ -26,44 +44,141 @@ export class CredentialStore implements ICredentialStore {
-                resolve(status);
-            }, err => reject(err));
-        });
+        try {
+            let success: boolean;
+            // Use native credential if the setting is on
+            if (this._useNativeCredentials) {
+                success = await this.nativeDeleteCredential(credentialId);
+            } else {
+                success = await this._client.sendRequest(Contracts.DeleteCredentialRequest.type, cred);
+            }
+            return success;
+        } catch (err) {
+            throw(err);
+        }
+    }
+
+    // Native Credential implementations
+
+    private async nativeReadCredential(credentialId: string, cred: Contracts.Credential): Promise<Contracts.Credential> {
+        try {
+            const savedPassword: string = await this._context.secrets.get(credentialId);
+            cred.password = savedPassword;
+            return cred;
+        } catch (err) {
+            throw(err);
+        }
+    }
+
+    private async nativeSaveCredential(credentialId: string, password: any): Promise<boolean> {
+        try {
+            await this._context.secrets.store(credentialId, password);
+            const savedPassword = await this._context.secrets.get(credentialId);
+            if (savedPassword === password) {
+                return true;
+            }
+            return false;
+        } catch (err) {
+            throw(err);
+        }
+    }
+
+    private async nativeDeleteCredential(credentialId: string): Promise<boolean> {
+        try {
+            await this._context.secrets.delete(credentialId);
+            const savedPassword = await this._context.secrets.get(credentialId);
+            if (!savedPassword) {
+                return true;
+            }
+            return false;
+        } catch (err) {
+            throw(err);
+        }
+    }
+
+    /**
+     *
+     * @returns Migrates all saved credentials to the native credential system
+     */
+    private async migratePasswords(): Promise<void> {
+        const connections = vscode.workspace.getConfiguration(Constants.extensionName).get<any[]>(Constants.connectionsArrayName);
+        const savedPasswordConnections = connections.filter(conn => conn.savePassword === true);
+        for (let i = 0; i < savedPasswordConnections.length; i++) {
+            let conn = savedPasswordConnections[i];
+            await this.cleanCredential(conn);
+        }
+        await Utils.removeCredentialFile();
+    }
+
+
+    private async cleanCredential(conn): Promise<boolean> {
+        const credentialId = ConnectionStore.formatCredentialId(conn);
+        const credential = await this._client.sendRequest(Contracts.ReadCredentialRequest.type, { credentialId, password: undefined });
+        if (credential.password) {
+            const password = credential.password;
+            // save it in secret store
+            await this._secretStorage.store(credentialId, password);
+            // check if it's saved
```

Comment:
Same as above, why do we need to check this? Once it's done being saved we should be fine to just delete it from STS.